### PR TITLE
User sees confusing toast in case with Server Required Authentication appears

### LINKED ISSUE #1778: User sees confusing toast in case with Server Required Authentication appears
#### Software and hardware versions 
Verified on Collect v1.11.1

#### Problem description
User sees confusing toast after specific steps when Server Required Authentication appears

#### Steps to reproduce the problem
1. User sees dialog:

2. User does not put credentials and clicks OK button
3. User sees the dialog again and clicks on Cancel button
4. User is moved to `Send Finalized Form` page and form is selected
5. User clicks on `Send Selected` button
6. User sees toast with information `Sorry, you have not selected any forms!`

#### Expected behavior
- Option 1: Form is unselected on `Send Finalized Form` page
- Option 2: ODK Collect detects that form is selected and confusing toast is not displayed

#### Other information 
Discovered during testing https://github.com/opendatakit/collect/pull/1768

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1778

#### What has been done to verify that this works as intended?
I tested the scenario mentioned from #1778

#### Why is this the best possible solution? Were any other approaches considered?
We should disable the button and clear the list.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@opendatakit-bot claim

**Comment by grzesiek2010** [comment]
@opendatakit-bot label "needs testing"

File: collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java [hunk 1]
```diff
@@ -111,7 +111,10 @@ public void onClick(View v) {
                     if (checkedItemCount > 0) {
                         // items selected
                         uploadSelectedFiles();
-                        InstanceUploaderList.this.listView.clearChoices();
+                        for (int i = 0; i < listView.getCount(); i++) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java]: You can use `setAllToCheckedState`.

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java]: Good point! thanks, I missed that method.

**Comment by mmarciniak90** [comment]
**Tested with success!**
Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.0.
Verified cases:
* Toast does not appear
* Forms are unselected
* Correct text on button is visible

**Comment by mmarciniak90** [comment]
@opendatakit-bot unlabel "needs testing"

**Review by lognaturel — APPROVED** [review]

**Review by lognaturel — APPROVED** [review]

**Pull request merged**